python/MOCK_sensors.py

```diff
@@ -122,8 +131,151 @@ def publish(self):
 
         if rospy.get_param('sensor_noise', default=False):
             data = self.add_noise(data)
+
+        brokenWindSensorStatus = rospy.get_param('broken_wind_sensors', default="not_broken")
+        if brokenWindSensorStatus != "not_broken":
+            rospy.logwarn("Broken Wind Sensor Status: " + brokenWindSensorStatus)
+
+        if (brokenWindSensorStatus == "stuck_at_zero"):
+            # This setting chooses a wind sensor which reads an angle of 0 and speed 0
+            # brokenSensorData is used to store an integer 1, 2, or 3 representing which sensor is broken
+
+            if(self.brokenSensorData == "uninitialized"):
+                # set up brokenSensorData so publish(self) remembers which sensor is stuck at zero
+                randomSeed = get_rosparam_or_default_if_invalid('random_seed', default=None, rosparam_type_cast=str)
+                random.seed(randomSeed)
+                self.brokenSensorData = random.randint(1, 3)
+
+            self.corrupt_wind_sensor(data, self.brokenSensorData, "angle", 0)
+            self.corrupt_wind_sensor(data, self.brokenSensorData, "speed", 0)
+
+            rospy.logwarn("MOCK wind_sensor_{} publishing corrupt data: ".format(self.brokenSensorData)
+                          + "angle and speed stuck at 0")
+
+        elif (brokenWindSensorStatus == "stuck_at_rand_constant"):
+            # This setting chooses a wind sensor which reads a constant random speed and direction
+            # brokenSensorData is a list containing: brokenSensorID, the constant angle, the constant direction.
+
+            if(self.brokenSensorData == "uninitialized"):
+                randomSeed = get_rosparam_or_default_if_invalid('random_seed', default=None, rosparam_type_cast=str)
+                random.seed(randomSeed)
+
+                brokenSensorID = random.randint(1, 3)
+                brokenSensorAngle_deg = random.random() * 360
+                brokenSensorSpeed_kts = random.random() * MAX_BROKEN_WINDSPEED_KTS
+                self.brokenSensorData = [brokenSensorID, brokenSensorAngle_deg, brokenSensorSpeed_kts]
```

Comment:
- Document `self.brokenSensorData` contents for each case
- Might be clearer to make it a dictionary

Reply:
Access `random.randint(1, 3)` with `self.brokenSensorData['brokenSensorID']`